chain-factory/src/factory.rs

```diff
@@ -66,14 +148,43 @@ pub trait FactoryModule {
         }
     }
 
+    fn get_deploy_chain_config_args(
+        &self,
+        min_validators: &usize,
+        max_validators: &usize,
+        min_stake: &BigUint,
+        additional_stake_required: &MultiValueEncoded<StakeMultiArg<Self::Api>>,
+    ) -> ManagedArgBuffer<Self::Api> {
+        let mut args = ManagedArgBuffer::new();
+
+        args.push_arg(min_validators);
+        args.push_arg(max_validators);
+        args.push_arg(min_stake);
+        args.push_multi_arg(additional_stake_required);
+
+        args
+    }
+
     #[only_owner]
     #[endpoint(blacklistSovereignChainSc)]
-    fn blacklist_sovereign_chain_sc(&self, sc_address: ManagedAddress) {
-        let _ = self.all_deployed_contracts().swap_remove(&sc_address);
+    fn blacklist_sovereign_chain_sc(&self, chain_id: ManagedBuffer, sc_address: ManagedAddress) {
+        let _ = self
+            .all_deployed_contracts(chain_id)
+            .swap_remove(&sc_address);
     }
 
-    #[proxy]
-    fn chain_config_proxy(&self) -> chain_config::Proxy<Self::Api>;
+    fn generate_chain_id(&self) -> ManagedBuffer {
+        let mut byte_array: [u8; 2] = [0; 2];
```

Comment:
These are the requirements? Isn't it too short?

Reply:
that's the requirement